pkg/registry/clusterrolebinding/policybased/storage.go

```diff
@@ -0,0 +1,59 @@
+/*
```

Comment:
@olegshaldybin yeah thanks for bring this up. This is precarious stuff trying to think of all the ways users can exploit this.

We could probably avoid a whole class of issues by not allow namespaced role bindings to reference cluster level roles. But I understand that this is wanted for convenience.